cmake/WasmBuildOptions.cmake

```diff
@@ -27,7 +27,6 @@ function(xeus_wasm_link_options target environment)
         PUBLIC "SHELL: -s EXPORT_NAME=\"createXeusModule\""
         PUBLIC "SHELL: -s EXPORT_ES6=0"
         PUBLIC "SHELL: -s USE_ES6_IMPORT_META=0"
-        PUBLIC "SHELL: -s DEMANGLE_SUPPORT=0"
```

Comment:
Deprecated so we would see this warnings while building kernels 
```
em++: warning: DEMANGLE_SUPPORT is deprecated (mangled names no longer appear in stack traces). Please open a bug if you have a continuing need for this setting [-Wdeprecated]
```
Noted here https://emscripten.org/docs/tools_reference/settings_reference.html#demangle-support